libs2eplugins/src/s2e/Plugins/Models/BaseFunctionModels.cpp

```diff
@@ -81,18 +82,29 @@ bool BaseFunctionModels::findNullCharWithWidth(S2EExecutionState *state, uint64_
 
         bool truth;
         bool res = solver->mustBeTrue(query, truth);
-        if (res && truth) {
+
+        if (truth && res) {
+            allSymbolic = false; // Found a null byte, not all characters are symbolic
+            getDebugStream(state) << "Found nullptr at offset " << len << "\n";
             break;
         }
     }
 
-    if (len == MAX_STRLEN) {
-        getDebugStream(state) << "Could not find nullptr char\n";
+    if (allSymbolic && len == MAX_STRLEN) {
+        // If we reached the end and all characters were symbolic, null-terminate the string
+        bool success = state->mem()->write(stringAddr + len - width, '\0'*width);
```

Comment:
`\0` * width is `0`,right?

Reply:
yes, you are right. I have updated the logic to create a buffer first and then pass the buffer to write